openproblems/tasks/_batch_integration/batch_integration_feature/datasets/batch_integration_graph.py

```diff
@@ -0,0 +1,32 @@
+from .....tools.decorators import dataset
+from .....tools.normalize import log_scran_pooling
+from ...batch_integration_graph.datasets.immune import immune_batch
+from ...batch_integration_graph.datasets.pancreas import pancreas_batch
+from typing import Callable
+
+
+def _convert_dataset_function(
+    func: Callable, name: str, image: str = "openproblems-r-base"
+) -> Callable:
+    metadata = func.metadata.copy()
+    metadata["image"] = image
+
+    @dataset(**metadata)
+    def converted_func(test=False):
+        adata = func(test=test)
+        adata = log_scran_pooling(adata)
+        adata.X = adata.layers["counts"]
+        adata.var_names_make_unique()
+        return adata
+
+    converted_func.__name__ = name
+
+    return converted_func
+
+
+immune_batch_log_scran = _convert_dataset_function(
+    immune_batch, "immune_batch_log_scran"
+)
+pancreas_batch_log_scran = _convert_dataset_function(
+    pancreas_batch, "pancreas_batch_log_scran"
+)
```

Comment:
Hmm I actually don't remember writing that function

Reply:
I think @scottgigante-immunai added that

Reply:
This function converts dataset functions from `batch_integration_graph` to be compatible with `batch_integration_feature`. Since `batch_integration_feature` [requires `adata.layers["log_scran_pooling"]](https://github.com/openproblems-bio/openproblems/pull/416/files/6df9300d899f905036beeba74daf1c7be2273f92#diff-db7ac95d5366fee40934c091dc2743cd245306ec830c8ed59a0e834fcb121217R13), we take the output of the graph dataset function and add the scran normalization to it. 

That said, I think this solution is too clever by half. Alternative is to do it manually:

```
@dataset(**dict(immune_batch.metadata, image="openproblems-r-base"))
def immune_batch_log_scran(test=False):
    adata = immune_batch(test=test)
    adata = log_scran_pooling(adata)
    adata.X = adata.layers["counts"]
    adata.var_names_make_unique()
    return adata


@dataset(**dict(pancreas_batch.metadata, image="openproblems-r-base"))
def pancreas_batch_log_scran(test=False):
    adata = pancreas_batch(test=test)
    adata = log_scran_pooling(adata)
    adata.X = adata.layers["counts"]
    adata.var_names_make_unique()
    return adata
```

which is maybe technically an undesirable repetition of code but it's easier to read.